core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactory.java

```diff
@@ -0,0 +1,359 @@
+package com.klaytn.caver.wallet.keyring;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.klaytn.caver.account.AccountKeyRoleBased;
+import com.klaytn.caver.account.WeightedMultiSigOptions;
+import com.klaytn.caver.utils.Utils;
+import org.web3j.crypto.*;
+import org.web3j.utils.Numeric;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.SignatureException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+/**
+ * Representing a KeyringFactory which supports create function for Keyring(Single/Multiple/RoleBased) instance.
+ */
+public class KeyringFactory {
+
+    /**
+     * Generates a keyring instance.
+     * @return Keyring
+     */
+    public static SingleKeyring generate() {
+        return KeyringFactory.generate(null);
+    }
+
+    /**
+     * Generates an keyring instance with entropy.
+     * @param entropy A random string to create keyring.
+     * @return Keyring
+     */
+    public static SingleKeyring generate(String entropy) {
```

Comment:
I think generateSingleKey, generateMultipleKey and generateRoleBasedKey should be implemented!